src/LanguageServer.php

```diff
@@ -65,6 +65,10 @@ public function __construct(ProtocolReader $reader, ProtocolWriter $writer)
     {
         parent::__construct($this, '/');
         $this->protocolReader = $reader;
+        $this->protocolReader->on('close', function () {
+            $this->shutdown();
+            $this->exit();
```

Comment:
At the moment, `shutdown` just frees the index to be garbage collected and `exit` will exit the process, so after that nothing can happen anymore